tools/releaser/main.go

```diff
@@ -0,0 +1,289 @@
+		return err
+	}
+
+	// update WORKSPACE
+	if verbose {
+		fmt.Println("Writing new WORKSPACE")
+	}
+	directivelessWorkspace := workspaceWithoutDirectives.Bytes()
+	newWorkspace := make([]byte, 0, len(directivelessWorkspace))
+	newWorkspace = append(newWorkspace, directivelessWorkspace[:directiveStart]...)
+	newWorkspace = append(newWorkspace, workspaceDirectivesBuff.Bytes()...)
+	newWorkspace = append(newWorkspace, directivelessWorkspace[directiveStart:]...)
+
+	if err := os.WriteFile("WORKSPACE", newWorkspace, 0644); err != nil {
+		return err
+	}
+
+	// cleanup before final gazelle run
+	if verbose {
+		fmt.Println("Cleaning up temporary files")
+	}
+	if err := os.Remove(_tmpBzl); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running final gazelle run, and copying some language specific build files.")
+	}
+	finalizationCommands := []struct {
+		cmd  string
+		args []string
+	}{
+		{cmd: "bazel", args: []string{"run", "//:gazelle"}},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_imports"}},
+		{cmd: "cp", args: []string{"-f", "bazel-bin/language/proto/known_imports.go", "language/proto/known_imports.go"}},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_proto_imports"}},
+		{cmd: "cp", args: []string{"-f", "bazel-bin/language/proto/known_proto_imports.go", "language/proto/known_proto_imports.go"}},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_go_imports"}},
+		{cmd: "cp", args: []string{"-f", "bazel-bin/language/proto/known_go_imports.go", "language/proto/known_go_imports.go"}},
+	}
+	for _, c := range finalizationCommands {
+		if out, err := exec.CommandContext(ctx, c.cmd, c.args...).CombinedOutput(); err != nil {
+			fmt.Println(string(out))
+			return err
+		}
+	}
+	return nil
+}
+
+func getWorkspaceWithouthDirectives(workspaceBuffer, workspaceWithoutDirectives *bytes.Buffer) (int, error) {
+	var directiveStart int
+	var directiveFound bool
+	workspaceScanner := bufio.NewScanner(workspaceBuffer)
+	for workspaceScanner.Scan() {
+		if bytes.HasPrefix(workspaceScanner.Bytes(), []byte("# gazelle:repository go_repository")) {
+			directiveFound = true
+			continue
+		}
+		n, err := workspaceWithoutDirectives.Write(workspaceScanner.Bytes())
+		if err != nil {
+			return directiveStart, err
+		}
+		_, err = workspaceWithoutDirectives.Write([]byte("\n"))
+		if err != nil {
+			return directiveStart, err
+		}
+		if !directiveFound {
+			directiveStart += n + 1
+		}
+	}
+
+	return directiveStart, nil
+}
+
+func getBuffsFromTmp(depsMaybeBuff, workspaceDirectivesBuff *bytes.Buffer) error {
+	tmpbzl, err := os.ReadFile(_tmpBzl)
+	if err != nil {
+		return err
+	}
+
+	attributeRegex := regexp.MustCompile(`^\s+(name|importpath) = "(.+)",$`)
+
+	var foundDef bool
+	var name string
+	tmpscanner := bufio.NewScanner(bytes.NewBuffer(tmpbzl))
+	for tmpscanner.Scan() {
+		if bytes.HasPrefix(tmpscanner.Bytes(), []byte("def ")) {
```

Comment:
```suggestion
		if bytes.HasPrefix(tmpscanner.Bytes(), []byte("def gazelle_dependencies")) {
```
To be more strict. You can also extract `gazelle_dependencies` to a string and use it here and the `-to_macro` flag above.

You can also consider creating a variable `currentLine = tmpscanner.Text()`, and use `strings.HasPrefix`, `strings.TrimSpace`, etc., so you don't need to keep creating byte slices from strings below.